### PR TITLE
Fix: Make Mentions Interactive with User Detail Modal

### LINKED ISSUE #679: Mentions in Chat Messages Are Not Clickable and not displaying User Profiles
### Description:

Mentions (@username) in the chat messages do not redirect or display their details when clicked. Instead, the mention appears as plain text without any interactive functionality.

### Steps to reproduce:

1. Go to the EmbeddedChat channel
2. Type a message containing a mention, e.g., @testuser, and send it.
3. Click on the mentioned username in the chat window.

### Expected behavior:
Clicking on a mentioned username (`@testuser`) should open a modal with the user's details.

https://github.com/user-attachments/assets/0d4b61f9-9bf5-44d6-80ce-1345715b728c

### Actual behavior:

The mention appears as plain text, and clicking on it does not trigger any action or provide user information.
Although it shows the user info on clicking the profile icon next to their message

https://github.com/user-attachments/assets/1a0eb136-1ddb-4380-8d81-e528add249ee

### LINKED ISSUE #689: Mentions in Chat messages do not display user profile on Click
### Description:

Mentions (@username) in the chat messages do not redirect or display their details when clicked. Instead, the mention appears as plain text without any interactive functionality.

### Steps to reproduce:

1. Go to the EmbeddedChat channel
2. Type a message containing a mention, e.g., @testuser, and send it.
3. Click on the mentioned username in the chat window.

### Expected behavior:
Clicking on a mentioned username (`@testuser`) should open a modal with the user's details.

https://github.com/user-attachments/assets/0d4b61f9-9bf5-44d6-80ce-1345715b728c

### Actual behavior:

The mention appears as plain text, and clicking on it does not trigger any action or provide user information.
Although it shows the user info on clicking the profile icon next to their message

https://github.com/user-attachments/assets/1a0eb136-1ddb-4380-8d81-e528add249ee

### PR DESCRIPTION
**Pull request by SinghaAnirban005**
# Brief Title
Make Mentions Interactive with User Detail Modal

## Acceptance Criteria fulfillment

- [x] Mentions in the chat input now trigger the user detail modal when clicked.

Fixes #689 

## Video/Screenshots

https://github.com/user-attachments/assets/59a81a77-4ee5-4210-b360-dc2fc4c6d67e

## PR Test Details

**Note**: The PR will be ready for live testing at https://rocketchat.github.io/EmbeddedChat/pulls/pr-680 after approval. Contributors are requested to replace `<pr_number>` with the actual PR number.

### CODE REVIEW

**Comment by devanshkansagra** [comment]
Hey @SinghaAnirban005, Please solve format issues

**Comment by devanshkansagra** [comment]
Hey @SinghaAnirban005 instead of installing prettier package, run yarn run format at the root of this project and then check for any linting issues using yarn run lint. also discard this new commit

File: packages/api/src/EmbeddedChatApi.ts [hunk 1]
```diff
@@ -1124,4 +1124,21 @@ export default class EmbeddedChatApi {
     const data = response.json();
     return data;
   }
+
+  async userData(username: string) {
```
Comment (Spiral-Memory) [inline packages/api/src/EmbeddedChatApi.ts:1128]: I believe the function to fetch `userDetails` already exists, if I remember correctly. Previously, clicking on the user avatar would display the user details in the sidepane. Is there a specific reason why this is needed again?

Comment (SinghaAnirban005) [reply packages/api/src/EmbeddedChatApi.ts:1128]: On clicking the user avatar we get access to the `user's _id ` but the similar thing is not available on clicking the mention. I could only get access to the user's `username` . Also the rocket chat's implementation of user info modal utilizes the api call via username, so i decided to go with the same approach. Thus , now we have two api's (1) which allows fetching userInfo through _id (2) allows fetching info through username

Comment (Spiral-Memory) [reply packages/api/src/EmbeddedChatApi.ts:1128]: Okay cool, understood

File: packages/react/src/views/UserInformation/UserInformation.js [hunk 2]
```diff
@@ -39,7 +39,7 @@ const UserInformation = () => {
   useEffect(() => {
     const getCurrentUserInfo = async () => {
       try {
-        const res = await RCInstance.userInfo(currentUser._id);
+        const res = await RCInstance.userData(currentUser.username);
```
Comment (Spiral-Memory) [inline packages/react/src/views/UserInformation/UserInformation.js:42]: username is present always right ?

Comment (SinghaAnirban005) [reply packages/react/src/views/UserInformation/UserInformation.js:42]: Yes

Comment (Spiral-Memory) [reply packages/react/src/views/UserInformation/UserInformation.js:42]: Ok

**Comment by thepiyush-303** [comment]
@SinghaAnirban005 
it shows userData is not a function to me.
can you help me.

**Comment by SinghaAnirban005** [comment]
@thepiyush-303 
1. In packages/api , run `yarn build`
2. then run the react project

**Comment by thepiyush-303** [comment]
Thanks @SinghaAnirban005,
It worked now.

**Pull request merged**